addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -203,6 +203,7 @@ export class Many2XAutocomplete extends Component {
         autocomplete_container: { type: Function, optional: true },
         dropdown: { type: Boolean, optional: true },
         autofocus: { type: Boolean, optional: true },
+        mapRecord: { type: Function, optional: true },
```

Comment:
suggestion to prevent duplicating the mapRecord logic:
instead of mapRecord, have a getOptionClassnames props, which takes a what's output by mapRecordToOption in arguments, and returns the classnames to apply to the option